Fix hyprctl Color and Icon problem

ysekka opened the pull request:
I have told about on color problem before. They color codes had to be written in hex form, not rgb form.
I also found that there is no no-icon option. So I fix it too.

yavko commented on src/ctl.rs:
```diff
@@ -172,9 +172,11 @@ pub mod set_error {
 pub mod notify {
     use super::*;
     use std::time::Duration;
-    #[repr(u8)]
```
Change to `i8` instead of removing

yavko commented on src/ctl.rs:
```diff
@@ -217,10 +219,38 @@ pub mod notify {
+
+/// An enum for special colors of notify
+#[derive(Copy, Clone)]
+pub enum NColor {
```
Instead of creating a new color type could you just make `Color` display with hex by default?

yavko commented on src/shared.rs:
```diff
@@ -188,8 +188,8 @@ impl Address {
-/// This pub(crate) function is used to write a value to a socket and to get the response
-pub(crate) async fn write_to_socket(
+/// This pub function is used to write a value to a socket and to get the response
+pub async fn write_to_socket(
```
Why are you making these public?

yavko commented on src/shared.rs:
```diff
@@ -215,8 +215,8 @@ pub(crate) async fn write_to_socket(
-/// This pub(crate) function is used to write a value to a socket and to get the response
-pub(crate) fn write_to_socket_sync(path: String, content: CommandContent) -> crate::Result<String> {
+/// This pub function is used to write a value to a socket and to get the response
+pub fn write_to_socket_sync(path: String, content: CommandContent) -> crate::Result<String> {
```
Why are you making these public?

yavko commented on src/shared.rs:
```diff
@@ -239,15 +239,15 @@ pub(crate) fn write_to_socket_sync(path: String, content: CommandContent) -> cra
-/// This pub(crate) enum holds the different sockets that Hyprland has
-pub(crate) enum SocketType {
+/// This pub enum holds the different sockets that Hyprland has
+pub enum SocketType {
```
Why are you making these public?

yavko commented on src/shared.rs:
```diff
@@ -239,15 +239,15 @@ pub(crate) fn write_to_socket_sync(path: String, content: CommandContent) -> cra
-/// This pub(crate) function gets the Hyprland socket path
-pub(crate) fn get_socket_path(socket_type: SocketType) -> String {
+/// This pub function gets the Hyprland socket path
+pub fn get_socket_path(socket_type: SocketType) -> String {
```
Why are you making these public?

yavko commented on src/shared.rs:
```diff
@@ -262,7 +262,7 @@ pub(crate) fn get_socket_path(socket_type: SocketType) -> String {
 }
 
-pub(crate) fn object_empty_as_none<'de, D, T>(deserializer: D) -> Result<Option<T>, D::Error>
+pub fn object_empty_as_none<'de, D, T>(deserializer: D) -> Result<Option<T>, D::Error>
```
Why are you making these public?

yavko replied:
Please keep this private, this is a very specific implementation detail that should not pollute the api

yavko commented on src/lib.rs:
```diff
@@ -65,4 +65,4 @@ pub(crate) mod unix_async {
 
 /// This type provides the result type used everywhere in Hyprland-rs
-pub type Result<T> = std::result::Result<T, shared::HyprError>;
+pub type Result<T> = std::result::Result<T, shared::HyprError>;
```
Why is this line changed?

ysekka replied:
I don't know what happened with this line. Actually nothing had changed. I have no Idea to how fix it, neither there is no problem with it. as there is nothing had changed in this line I think there is no problem with it.

yavko reviewed (changes requested):
I understand that the notify command might not work with decimal number, but please keep the PR on topic, why are so many functions made public?

ysekka commented:
Someone maybe wants to use those functions. For example, there is a guy who forked Hyprland and added some commands to it wants to interact with it by rust. That is clear that we cannot add every single commands of a hyprland fork to hyprland-rs. If someone wants to execute a custom command, he/she can do this. If a function which execute custom commands exists, I can make them pub(crate) again.

yavko commented on src/shared.rs:
```diff
@@ -188,7 +188,7 @@ impl Address {
 }
 
-/// This pub(crate) function is used to write a value to a socket and to get the response
+/// This pub function is used to write a value to a socket and to get the response
```
Where did the `(crate)` go? Please fix this for all of them

yavko commented on src/shared.rs:
```diff
@@ -239,14 +239,14 @@ pub(crate) fn write_to_socket_sync(path: String, content: CommandContent) -> cra
     Listener,
 }
-/// This pub(crate) function gets the Hyprland socket path
+/// This pub function gets the Hyprland socket path
```
here too

yavko commented:
> Someone maybe wants to use those functions. For example, there is a guy who forked Hyprland and added some commands to it wants to interact with it by rust. That is clear that we cannot add every single commands of a hyprland fork to hyprland-rs. If someone wants to execute a custom command, he/she can do this. If a function which execute custom commands exists, I can make them pub(crate) again.

If they fork the repo, the functions will be available that's what `pub(crate)` means, it means it's available to the current crate? And I'm planning on introducing a better way of doing this that's more user oriented, so please reverse that. Also, please fix your commit names, this is a public project, such vague commit messages are not allowed. Preferably, make them into one through like a force push or something.

yavko commented on src/ctl.rs:
```diff
@@ -217,8 +220,8 @@ pub mod notify {
-#[derive(MDisplay, Constructor)]
-#[display(fmt = "rgba({},{},{},{})", "_0", "_1", "_2", "_3")]
+#[derive(Copy, Clone, MDisplay, Constructor)]
+#[display(fmt = "rgba({}{}{}{})", "_0", "_1", "_2", "_3")]
```
You never changed this to hex?

ysekka commented:
I converted all leftover pub's to pub(crate)s. I also fix the hex conversion and I replaced i8 with i32. There is no change occurred in src/lib.rs btw. I hope that I hadn't left anything.

yavko commented:
The commit names are still not fixed
